api/v1alpha1/envoyproxy_types.go

```diff
@@ -115,6 +120,25 @@ type EnvoyProxyProvider struct {
 	Kubernetes *EnvoyProxyKubernetesProvider `json:"kubernetes,omitempty"`
 }
 
+// ShutdownConfig defines configuration for graceful envoy shutdown process.
+type ShutdownConfig struct {
+	// DrainTimeout defines the graceful shutdown timeout. This should be less than the pod's terminationGracePeriodSeconds.
+	// If unspecified, defaults to 600 seconds.
+	//
+	// +optional
+	DrainTimeout *metav1.Duration `json:"drainTimeout,omitempty"`
+	// MinDrainDuration defines the minimum drain duration allowing time for endpoint deprogramming to complete.
+	// If unspecified, defaults to 15 seconds.
+	//
+	// +optional
```

Comment:
can we do without `MinDrainDuration` & `ExitAtConnections` in the first iteration ?

Reply:
Removed `ExitAtConnections` in fb4e3ed but `MinDrainDuration` really should stay. It's an important knob that should be configurable. I will be using it when rolling this out to infra I manage, and it's being used in `test/config/gatewayclass.yaml` to prevent a conservative default of 15 seconds from slowing CI processes.

Reply:
can you elaborate a little more on the endpoint deprogramming part ? this timeout is added to ensure the proxy's VIP is removed from kube-proxy or other external loadbalancers ?

Reply:
if this value is present to deal with eventual consistency of the various distributed networking components, prefer if we dropped this to 5s

Reply:
> to deal with eventual consistency of the various distributed networking components

That's exactly what it's for. I've lowered the default to 5s